src/main/java/nextstep/ladder/domain/line/Line.java

```diff
@@ -1,35 +1,65 @@
 package nextstep.ladder.domain.line;
 
-import java.util.ArrayList;
 import java.util.List;
-import nextstep.ladder.domain.RandomGenerator;
+import java.util.stream.IntStream;
+import nextstep.ladder.error.exception.LineNoConsecutiveTrues;
 
 public class Line {
 
-    private List<LadderConstructionStatus> points = new ArrayList<>();
+    private final List<Point> points;
 
-    public Line(int countOfPerson) {
-        initializeLine(countOfPerson);
+    public Line(LadderConnectOrder ladderConnectOrder) {
+        this(initializeLine(ladderConnectOrder));
     }
 
-    private void initializeLine(int countOfPerson) {
-        LadderConstructionStatus ladderConstructionStatus = new LadderConstructionStatus(false);
-        for (int count = 0; count < countOfPerson - 1; count++) {
-            ladderConstructionStatus = addLineValue(ladderConstructionStatus);
+    public Line(List<Point> points) {
+        if (isConsecutiveTrues(points)) {
+            throw new LineNoConsecutiveTrues();
         }
+        this.points = points;
     }
 
-    private LadderConstructionStatus addLineValue(
-        LadderConstructionStatus previousLadderConstructionStatus) {
-        LadderConstructionStatus newStatus =
-            previousLadderConstructionStatus.isLadderConstructionStatusFalse()
-                ? new LadderConstructionStatus(RandomGenerator.createRandomBoolean())
-                : new LadderConstructionStatus(false);
-        points.add(newStatus);
-        return newStatus;
+    private static List<Point> initializeLine(LadderConnectOrder ladderConnectOrder) {
+        return ladderConnectOrder.connectLadder();
     }
 
-    public List<LadderConstructionStatus> getPoints() {
+
+    private boolean isConsecutiveTrues(List<Point> points) {
+        return IntStream.range(0, points.size() - 1)
+            .anyMatch(i -> points.get(i).isPointTrue() && points.get(i + 1).isPointTrue());
+    }
+
+    public boolean canMoveRight(int userIndex) {
+        if (isValidUserIndexSize(userIndex) && isRightTrue(userIndex)) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean canMoveLeft(int userIndex) {
+        if (isValidUserIndexPositive(userIndex) && isLeftTrue(userIndex)) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isValidUserIndexSize(int userIndex) {
+        return userIndex < points.size() ? true : false;
+    }
```

Comment:
삼항연산을 할 필요가 없을 것 같아요 
비슷한 메서드가 여럿있네요 
```suggestion
    private boolean isValidUserIndexSize(int userIndex) {
        return userIndex < points.size();
    }
```

Reply:
앗 그렇네요
사다리타기 도메인에 집중한 나머지 이상한 코드가 나와버렸네요 😭